presto-spi/src/main/java/com/facebook/presto/spi/block/Block.java

```diff
@@ -207,7 +207,21 @@ default Block getPositions(int[] positions)
      * <p>
      * The returned block must be a compact representation of the original block.
      */
-    Block copyPositions(List<Integer> positions);
+    Block copyPositions(int[] positions, int offset, int length);
```

Comment:
consider adding

```
    default Block copyPositions(int[] positions) {
        return copyPositions(positions, 0, positions.length);
    }
```

There are a lot of such usages in subsequent commits.

Reply:
True, but we mostly use array builder and it's really easy to incorrectly call `copyPositions(in[] postiions)` by mistake instead of `copyPositions(int[] positions, int offset, int length);`. Because of that I prefer to leave only extended, explicit version of the method.

Reply:
Extended version requires boilerplate and requires an array to be on a variable to make a call. See `DictionaryBlock` constructors for reference :)

Reply:
Shortened versions are used mostly in tests, while extend version is used mostly (if not exclusively) in main code.

Reply:
Checked again, extended version would be used exclusively in main code. Therefore I think it should be only extended version of method in SPI.

Reply:
Actually when I first read the interface, it kinda confuses me with `offset` and `length`. I'm ok leaving the extended version only; how about explaining more in the java doc?

Reply:
added better javadoc